plugin/src/main/java/at/helpch/chatchat/channel/ChatChannel.java

```diff
@@ -41,25 +42,36 @@ public static void defaultChannel(@NotNull final ChatChannel toSet) {
     @Override
     public String toString() {
         return "ChatChannel{" +
-                "name=" + name() +
-                ", messagePrefix='" + messagePrefix() + '\'' +
-                ", toggleCommands='" + commandNames() + '\'' +
-                ", channelPrefix='" + channelPrefix() + '\'' +
-                ", radius='" + radius() +
-                '}';
+            "name=" + name() +
+            ", messagePrefix='" + messagePrefix() + '\'' +
+            ", toggleCommands='" + commandNames() + '\'' +
+            ", channelPrefix='" + channelPrefix() + '\'' +
+            ", radius='" + radius() +
+            '}';
     }
 
     @Override
     public Set<User> targets(final @NotNull User source) {
+
+        final Predicate<User> filterIgnores = user -> {
+            // console can't be ignored
+            if (source instanceof ConsoleUser) return true;
+            return !user.ignoredUsers().contains(source.uuid()) ||
+                ((ChatUser) source).player().hasPermission(IgnoreCommand.IGNORE_BYPASS_PERMISSION);
```

Comment:
We might want to split these 2 and add an extra check to see if the user is actually a ChatUser. People could add their own users (in the future :) ) so we might end up with some exceptions thrown here.

Reply:
Is the recent commit better? I think that's what you're asking for